### ISSUE TITLE
[BUG] datadog agent not starting on bottlerocket 1.9.2

### ISSUE THREAD

**Issue by willthames** [DataDog/datadog-agent#13447]
The system-probe container within the datadog agent does not start on bottlerocket 1.9.2 (it works fine with 1.9.0 beforehand, and downgrading the nodes back to 1.9.0 resolved the datadog agent failure to start)

I suspect this is related to the runc upgrade from 1.1.3 to 1.1.4 in bottlerocket 1.9.2

**Agent Environment**
```
Agent 7.38.2 - Commit: ba442fd - Serialization version: v5.0.23 - Go version: go1.17.11
```

`agent status` returns nothing

**Describe what happened:**
> failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec /opt/datadog-agent/embedded/bin/system-probe: operation not permitted: unknown

**Describe what you expected:**
system-probe starts

**Steps to reproduce the issue:**
Install datadog agent from helm chart (we're using 2.37.9 but 3.0.1 looks like it would behave identically) and add a bottlerocket 1.9.2 node to the cluster. 

**Additional environment details (Operating System, Cloud provider, etc):**
AWS
EKS 1.23
Bottlerocket 1.9.2

**Comment by contrarianchris** [comment]
Seeing the same behaviour...
```
Bottlerocket 1.9.2-b8074d44
Datadog Helm Chart 3.0.1
Datadog Agent 7.38.2
```

**Comment by willthames** [comment]
Looks like this is/will be fixed in the next version of datadog helm chart 

https://github.com/DataDog/helm-charts/pull/740

**Comment by willthames** [comment]
I’ll close this as it’s fixed in version 3.0.3

https://github.com/DataDog/helm-charts/compare/datadog-3.0.2...datadog-3.0.3
